lib/preview-web/src/PreviewWeb.test.ts

```diff
@@ -0,0 +1,2084 @@
+      await new PreviewWeb({
+        getProjectAnnotations: () => ({}),
+        importFn,
+        fetchStoryIndex,
+      }).initialize();
+
+      expect(mockChannel.emit).toHaveBeenCalledWith(Events.SET_GLOBALS, {
+        globals: {},
+        globalTypes: {},
+      });
+    });
+
+    it('emits the SET_GLOBALS event from the URL', async () => {
+      document.location.search = '?id=*&globals=a:c';
+
+      await new PreviewWeb({ getProjectAnnotations, importFn, fetchStoryIndex }).initialize();
+
+      expect(mockChannel.emit).toHaveBeenCalledWith(Events.SET_GLOBALS, {
+        globals: { a: 'c' },
+        globalTypes: {},
+      });
+    });
+
+    it('sets args from the URL', async () => {
+      document.location.search = '?id=component-one--a&args=foo:url';
+
+      const preview = new PreviewWeb({ getProjectAnnotations, importFn, fetchStoryIndex });
+      await preview.initialize();
+
+      expect(preview.storyStore.args.get('component-one--a')).toEqual({
+        foo: 'url',
+      });
+    });
+  });
+
+  describe('initial selection', () => {
+    it('selects the story specified in the URL', async () => {
+      document.location.search = '?id=component-one--a';
+
+      const preview = new PreviewWeb({ getProjectAnnotations, importFn, fetchStoryIndex });
+      await preview.initialize();
+
+      expect(preview.urlStore.selection).toEqual({
+        storyId: 'component-one--a',
+        viewMode: 'story',
+      });
+      expect(history.replaceState).toHaveBeenCalledWith(
+        {},
+        '',
+        'pathname?id=component-one--a&viewMode=story'
+      );
+    });
+
+    it('emits the STORY_SPECIFIED event', async () => {
+      document.location.search = '?id=component-one--a';
+
+      await new PreviewWeb({ getProjectAnnotations, importFn, fetchStoryIndex }).initialize();
+
+      expect(mockChannel.emit).toHaveBeenCalledWith(Events.STORY_SPECIFIED, {
+        storyId: 'component-one--a',
+        viewMode: 'story',
+      });
+    });
+
+    it('emits the CURRENT_STORY_WAS_SET event', async () => {
+      document.location.search = '?id=component-one--a';
+
+      await new PreviewWeb({ getProjectAnnotations, importFn, fetchStoryIndex }).initialize();
+
+      expect(mockChannel.emit).toHaveBeenCalledWith(Events.CURRENT_STORY_WAS_SET, {
+        storyId: 'component-one--a',
+        viewMode: 'story',
+      });
+    });
+
+    describe('if the story specified does not exist', () => {
+      it('renders missing', async () => {
+        document.location.search = '?id=random';
+
+        const preview = new PreviewWeb({ getProjectAnnotations, importFn, fetchStoryIndex });
+        await preview.initialize();
+
+        expect(preview.view.showNoPreview).toHaveBeenCalled();
+        expect(mockChannel.emit).toHaveBeenCalledWith(Events.STORY_MISSING, 'random');
+      });
+
+      // TODO we need tests for story list changes
+      it.skip('tries again with a specifier if CSF file changes', async () => {
```

Comment:
skip?

Reply:
These tests can't really be written properly until we have a mutable StoryIndex server, they are really just placeholders until then.